File: 1037-minimum-number-of-k-consecutive-bit-flips/1037-minimum-number-of-k-consecutive-bit-flips.c
int minKBitFlips(int* nums, int numsSize, int k) {
    int flip_count = 0;
    int flips[numsSize];
    int ans =0;
    for(int i=0; i<numsSize; i++){
        flips[i] = 0;
        if(i>=k){
            flip_count ^= flips[i-k];
        }
        if(nums[i]^flip_count == 0){
            if(i>numsSize-k) return -1;
            flip_count ^=1;
            flips[i] = 1;
            ans ++;
        }
    }
    return ans;
}

File: 1037-minimum-number-of-k-consecutive-bit-flips/1037-minimum-number-of-k-consecutive-bit-flips.py
class Solution:
    def minKBitFlips(self, nums: List[int], k: int) -> int:
        
        flip_count = 0
        flips = [0]*len(nums)
        ans =0
        for now in range(len(nums)):
            if now >= k:
                flip_count ^= flips[now-k]
            if nums[now]^flip_count == 0:
                if now > len(nums)-k:
                    return -1
                flip_count ^=1
                flips[now] = 1
                ans+=1
            
        
        return ans
        